lib/vector-core/src/fanout.rs

```diff
@@ -130,11 +130,30 @@ impl Fanout {
         }
     }
 
-    pub async fn send_stream(&mut self, events: impl Stream<Item = EventArray>) {
+    /// Send a stream of events to all connected sinks.
+    ///
+    /// This function will send events until the provided stream finishes. It will also block on the
+    /// resolution of any pending reload before proceeding with a send operation, similar to `send`.
+    ///
+    /// # Panics
+    ///
+    /// This method can panic if the fanout receives a control message that violates some invariant
+    /// about its current state (e.g. remove a non-existent sink, etc.). This would imply a bug in
+    /// Vector's config reloading logic.
+    ///
+    /// # Errors
+    ///
+    /// If an error occurs while sending events to any of the connected sinks, an error variant will be
+    /// returned detailing the cause.
+    pub async fn send_stream(
+        &mut self,
+        events: impl Stream<Item = EventArray>,
+    ) -> Result<(), Box<dyn error::Error + Send + Sync>> {
```

Comment:
Could this (and all following) use `crate::Error` or `crate::Result`?

Reply:
Totally could. 👍🏻

Reply:
Fixed this in ddd91a8d6.